schema.go

```diff
@@ -7,6 +7,7 @@ import (
 type schema interface {
 	unmarshal(evt *Event) error
 	timestamp() time.Time
+	hasCanaryEvent() bool
```

Comment:
Should we split this interface into individual interfaces and then combine it with composition?  Like we do with other interfaces?